bigcode_eval/utils.py

```diff
@@ -334,8 +335,9 @@ def complete_code(
                     gen_token_dict,
                 )
                 with open(intermediate_save_generations_path, "w") as fp:
-                    intermediate_generations.extend(code_gens)
-                    json.dump(intermediate_generations, fp)
+                    intermediate_save_generations = deepcopy(generations)
+                    intermediate_save_generations.extend(code_gens)
+                    json.dump(intermediate_save_generations, fp)
```

Comment:
Small suggestion:
```
json.dump(generations + code_gens, fp)
```
to avoid having a deepcopy.

Reply:
sure, can do!